README.md
zamger
======

Zamger https://zamger.etf.unsa.ba


doc/ChangeLog.txt
------------------------------------------------------------------------
r831 | vedran | 2010-07-13 23:39:50 +0200 (Tue, 13 Jul 2010) | 2 lines

Objavljujem v4.1.1

------------------------------------------------------------------------
r830 | vedran | 2010-07-13 23:29:44 +0200 (Tue, 13 Jul 2010) | 2 lines

(v4.1) Disablujem module za anketu

------------------------------------------------------------------------
r829 | vedran | 2010-07-13 23:23:32 +0200 (Tue, 13 Jul 2010) | 2 lines

(v4.1) Dodajem potrebne fajlove

------------------------------------------------------------------------
r825 | vedran | 2010-07-10 20:57:38 +0200 (Sat, 10 Jul 2010) | 2 lines

(v4.1) student/zadaca: Bolje rjesenje za prethodni problem

------------------------------------------------------------------------
r823 | vedran | 2010-07-10 20:24:06 +0200 (Sat, 10 Jul 2010) | 3 lines

(v4.1) student/zadaca: Dodatni escaping imena fajla prilikom slanja 
attachmenta

------------------------------------------------------------------------
r821 | vedran | 2010-07-10 03:30:04 +0200 (Sat, 10 Jul 2010) | 3 lines

(v4.1) studentska/osobe: Podesavanje tipa korisnika (za admina) je 
prestajalo raditi nakon par upotreba (bug otkrio Amer Zec)

------------------------------------------------------------------------
r816 | vedran | 2010-06-21 03:28:23 +0200 (Mon, 21 Jun 2010) | 3 lines

(v4.1) saradnik/student: Onemoguceno prebacivanje u istu grupu (bug 
#103), interfejs je blago modifikovan

------------------------------------------------------------------------
r815 | vedran | 2010-06-21 03:25:45 +0200 (Mon, 21 Jun 2010) | 3 lines

(v4.1) Ne prikazuj obavijest o zadaci ako slanje zadaca nije aktivirano 
u adminu.

------------------------------------------------------------------------
r814 | vedran | 2010-06-21 03:22:21 +0200 (Mon, 21 Jun 2010) | 2 lines

(v4.1) Kod roka za slanje zadace i vrijeme je bitno

------------------------------------------------------------------------
r813 | vedran | 2010-06-21 03:19:25 +0200 (Mon, 21 Jun 2010) | 2 lines

(v4.1) student/kolizija: Sprijeci SQL error kad je prazna baza

------------------------------------------------------------------------
r812 | vedran | 2010-06-21 03:16:35 +0200 (Mon, 21 Jun 2010) | 3 lines

(v4.1) izvjestaj/index: Popravljen bug sa visestrukim prikazivanjem 
istog predmeta

------------------------------------------------------------------------
r811 | vedran | 2010-06-21 03:13:49 +0200 (Mon, 21 Jun 2010) | 2 lines

(v4.1) rss.php: razne ispravke na osnovu sajta feedvalidator.org

------------------------------------------------------------------------
r785 | vedran | 2010-06-04 11:45:08 +0200 (Fri, 04 Jun 2010) | 3 lines

(v4.1) saradnik/grupa: ajah_box nije bio ispisivan ako na predmetu ne 
postoji nijedna "klasicna" komponenta za prisustvo

------------------------------------------------------------------------
r784 | vedran | 2010-06-04 11:43:07 +0200 (Fri, 04 Jun 2010) | 3 lines

(v4.1) izvjestaj/grupe: Zbog greske u nazivu varijable nije se ispisivao 
spisak studenata koji nisu ni u jednoj grupi

------------------------------------------------------------------------
r783 | vedran | 2010-06-04 11:41:54 +0200 (Fri, 04 Jun 2010) | 3 lines

(v4.1) nastavnik/grupe: Kod masovnog upisa, za studente koji nisu 
upisani ni u jednu grupu pisalo "Student vec upisan u grupu '' "

------------------------------------------------------------------------
r777 | vedran | 2010-03-14 04:00:03 +0100 (Sun, 14 Mar 2010) | 4 lines

(v4.1) common/ajah: U akciji izmjena_ispita (tip "ispit") ponuda kursa 
nije bila dobro odredjena u slucaju vise ponuda istog kursa, uzimana je 
random ponuda kursa umjesto one koju student slusa

------------------------------------------------------------------------
r776 | vedran | 2010-03-14 03:57:06 +0100 (Sun, 14 Mar 2010) | 7 lines

(v4.1) izvjestaj/statistika_predmeta: Pored poznatog problema sa 
studentom koji je izasao na samo jedan ispit, izvjestaj je davao netacne 
rezultate i ako je odrzano vise ispita istog tipa. Privremeno "hack" 
rjesenje preskace komponentu koja se zove Usmeni (sto nece raditi ako se 
ispit preimenuje u npr. Zavrsni ispit). Moguce rjesenje je definisanje 
uslova za izlazak na ispit sto se moze uvezati sa prijavom ispita.

------------------------------------------------------------------------
r775 | vedran | 2010-03-14 03:56:13 +0100 (Sun, 14 Mar 2010) | 2 lines

(v4.1) studentska/izvjestaji: Popravljen prikaz prozora za cekanje na IE8

------------------------------------------------------------------------
r774 | vedran | 2010-03-14 03:55:25 +0100 (Sun, 14 Mar 2010) | 5 lines

(v4.1) nastavnik/grupe: Izbjegnuta SQL greska za slucaj da se student 
dva puta upisuje u istu grupu. Dodana poruka za slucaj ispisa i upisa u 
istu grupu (ali se to ne uzima kao greska). Popravljen ispis kod upisa u 
vise grupa istovremeno.

------------------------------------------------------------------------
r773 | vedran | 2010-03-14 03:53:35 +0100 (Sun, 14 Mar 2010) | 3 lines

(v4.1) studentska/osobe: Ponekad nije ispravno detektovano da je student 
upisan u parni semestar tekuce ak. godine.

------------------------------------------------------------------------
r772 | vedran | 2010-03-14 03:52:21 +0100 (Sun, 14 Mar 2010) | 3 lines

(v4.1) studentska/obavijest: Zbog greske u SQLu nije radilo slanje maila 
svim studentima na godini

------------------------------------------------------------------------
r771 | vedran | 2010-03-14 03:51:17 +0100 (Sun, 14 Mar 2010) | 3 lines

(v4.1) index.php: Dodan link na dokumentaciju na sve stranice. Popravljen 
prikaz nove poste na IE8

------------------------------------------------------------------------
r770 | vedran | 2010-03-14 03:50:30 +0100 (Sun, 14 Mar 2010) | 2 lines

(v4.1) student/prijava_ispita: Sortiraj ispite po datumu

------------------------------------------------------------------------
r769 | vedran | 2010-03-14 03:49:32 +0100 (Sun, 14 Mar 2010) | 3 lines


index.php
<?


// INDEX - master skripta za ZAMGER

// v3.9.1.0 (2008/02/09) + Pocetak rada
// v3.9.1.1 (2008/03/08) + Ukinute uloge
// v3.9.1.2 (2008/03/21) + Popravljen pristup public dijelovima bez prijave i logging
// v3.9.1.3 (2008/04/17) + Popravljen XSS sa varijablom sta
// v3.9.1.4 (2008/05/07) + Dodan logging za razne greske kod anonimnog pristupa, popravljen forwarding kada čovjeku istekne sesija
// v3.9.1.5 (2008/08/27) + Koristimo horizontalni_meni() za studentska/*; koristimo tabelu privilegija umjesto auth
// v3.9.1.6 (2008/10/02) + Popravljen logging
// v4.0.0.0 (2009/02/19) + Release
// v4.0.0.1 (2009/04/01) + Dodan link na RSS u header
// v4.0.9.1.(2009/06/16) + Dodan link na dokumentaciju na sve stranice


require("lib/libvedran.php");
require("lib/zamger.php");
require("lib/config.php");

dbconnect2($conf_dbhost,$conf_dbuser,$conf_dbpass,$conf_dbdb);


// Login forma i provjera sesije

$greska="";
$sta = my_escape($_REQUEST['sta']);

// Ovaj kod smo ukinuli da bi se moglo sa login stranice redirektovati tamo gdje je korisnik već bio
//if ($_REQUEST['greska']==1) {
	// Da li ce se ovo ikada desiti?
//	$greska="Vaša sesija je istekla. Molimo prijavite se ponovo.";
//}

if ($_POST['loginforma'] == "1") {
	$login = my_escape($_POST['login']);
	$pass = my_escape($_POST['pass']);
	
	if (!preg_match("/[\w\d]/",$login)) {
		$greska="Nepoznat korisnik";
	} else {
		$status = login($pass);
		if ($status == 1) { 
			$greska="Nepoznat korisnik";
		} else if ($status == 2) {
			$greska="Pogrešna šifra";
		} 
	}
	if ($greska=="") zamgerlog("login",1); // nivo 1 = posjeta stranici

} else {
	check_cookie();
//	$userid=0;
	if ($userid==0 && $sta!="" && $sta!="public/intro") {
		$greska = "Vaša sesija je istekla. Molimo prijavite se ponovo.";
//		$sta = ""; // -- Ne brisati sta! treba za provjeru public pristupa
	}
}

// nakon dijela iznad, $userid drzi numericki ID prijavljenog korisnika


// SU = switch user

if ($userid>0) {
	$su = intval($_REQUEST['su']);
	if ($su==0) $su = intval($_SESSION['su']);
	$unsu = intval($_REQUEST['unsu']);
	if ($unsu==1 && $su!=0) $su=0;
	if ($su>0) {
		// Provjeravamo da li je korisnik admin
		$q5 = myquery("select count(*) from privilegije where osoba=$userid and privilegija='siteadmin'");
		if (mysql_result($q5,0,0)>0) {
			$userid=$su;
			$_SESSION['su']=$su;
		} 
	} else {
		$_SESSION['su']="";
	}
}




// Određivanje privilegija korisnika

$user_student=$user_nastavnik=$user_studentska=$user_siteadmin=false;
if ($userid>0) {
	$q10 = myquery("select privilegija from privilegije where osoba=$userid");
	while ($r10=mysql_fetch_row($q10)) {
		if ($r10[0]=="student") $user_student=true; 
		if ($r10[0]=="nastavnik") $user_nastavnik=true;
		if ($r10[0]=="studentska") $user_studentska=true;
		if ($r10[0]=="siteadmin") $user_siteadmin=true;
		//if ($r10[0]=="prijemni")  -- ovi nemaju pristup zamgeru
		// ovdje dodati ostale vrste korisnika koje imaju pristup
	}


	// Korisnik nije ništa!?
	if (!$user_student && !$user_nastavnik && !$user_studentska && !$user_siteadmin) {
		$greska = "Vaše korisničko ime je ispravno, ali nemate nikakve privilegije na sistemu! Kontaktirajte administratora.";
		$sta = "";
	}
}




// Pronalazenje trazenog modula u registryju

include("registry.php");
$staf = str_replace("/","_",$sta);
$found=0;
$naslov="";
if ($sta!="") { // Ne kontrolisemo gresku, zbog public pristupa
	// Logout
	if ($sta == "logout") {
		logout();
		zamgerlog("logout",1);
		$userid=0;
		$sta="public/intro";
		$staf="public_intro";
		$found=1;
	}

	// Pretraga
	foreach ($registry as $r) {

		if ($r[0] == $sta) { //$r[5] == debug
			if (strstr($r[3],"P") || (strstr($r[3],"S") && $user_student) || (strstr($r[3],"N") && $user_nastavnik) || (strstr($r[3],"B") && $user_studentska) || (strstr($r[3],"A") && $user_siteadmin)) {
				$naslov=$r[1];
				$template=$r[4];
				$found=1;
				$greska = "";
			} else if ($greska=="") {
				$greska = "Pristup nije dozvoljen";
				$permstr=""; // opis korisnika, za lakši debugging
				if ($user_student) $permstr.="S";
				if ($user_nastavnik) $permstr.="N";
				if ($user_studentska) $permstr.="B";
				if ($user_siteadmin) $permstr.="A";
				if ($userid>0) zamgerlog("Korisnik $userid (tip $permstr) pokusao pristupiti $sta sto zahtijeva $r[3]",3); // nivo 3 = greska
				$sta = ""; // prikaži default modul
//print "Korisnik $userid (tip $permstr) pokusao pristupiti $sta sto zahtijeva $r[3]";
			} else {
				$sta=""; // kako se ne bi prikazivale ostale greske, navigacija itd.
			}
			break;
		}
	}
}

if ($naslov=="") $naslov = "ETF Bolognaware"; // default naslov

// template==2 - ne prikazujemo ni header (npr. PDF ispis)
if ($found==1 && $template==2 && $greska=="") {
	if ($userid>0) zamgerlog(urldecode(genuri()),1); // nivo 1 = posjet stranici
	include ("$sta.php");
	eval("$staf();");
	exit;
}


?>
<html>
<head>
	<title><?=$naslov?></title>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<link href="css/zamger.css" rel="stylesheet" type="text/css" />
	<link rel="alternate" type="application/rss+xml" title="RSS 2.0" href="http://zamger.etf.unsa.ba/rss.php" />
</head><?


// Prikaz modula bez template-a

if ($found==1 && $template==0 && $greska=="") {
	// Greske uvijek prikazujemo u template-u
	print "<body bgcolor=\"#FFFFFF\">\n";
	if ($userid>0) zamgerlog(urldecode(genuri()),1); // nivo 1 = posjet stranici
	include ("$sta.php");
	eval("$staf();");
	print "</body></html>\n";
	exit;
}


// Slijedi template

?>
<body topmargin="0" leftmargin="0" bottommargin="0" rightmargin="0" bgcolor="#FFFFFF">

<script type="text/javascript" src="js/stablo.js"></script> <!-- Cesto koristena skripta -->

<table width="100%" border="0" cellspacing="0" cellpadding="0">
	<tr bgcolor="#BBBBFF">
		<!--td><table width="100%" border="0" cellspacing="0" cellpadding="0"><tr-->
			<td width="50%">&nbsp;&nbsp;&nbsp;&nbsp;
			<a href="index.php"><img src="images/etf-50x50.png" width="50" height="50" border="0"></a>
			</td><td width="50%" align="right">
			<font color="#FFFFFF" size="5">
			<b><a href="index.php"><font color="#FFFFFF"><?=$conf_appname?> <?=$conf_appversion?></font></a>&nbsp;</b></font><br/>
			<font color="#FFFFFF" size="1">
			<a href="doc/zamger-uputstva-40-nastavnik.pdf" target="_new">
			<img src="images/16x16/dokumentacija.png" width="16" height="16" border="0" align="center">&nbsp;
			Uputstva</a>&nbsp;&nbsp;&nbsp;
			<a href="http://195.130.59.135/bugzilla" target="_new">
			<img src="images/16x16/zad_bug.png" width="16" height="16" border="0" align="center">&nbsp;
			Prijavite bug</a>&nbsp;&nbsp;&nbsp;</font>
			</td>
		<!--/tr></table></td-->
	</tr>
</table>

<table width="100%" border="0" cellspacing="0" cellpadding="4">
	<tr><td valign="top" align="left">

<?

// Provjera maila
if ($userid>0) {
	$q20 = myquery("select UNIX_TIMESTAMP(vrijeme) from log where userid=$userid order by id desc limit 2");
	if (mysql_num_rows($q20)>0)
		$vrijeme=intval(mysql_result($q20,1,0));
	else 
		$vrijeme=0;
	$q30 = myquery("select count(*) from poruka where tip=2 and opseg=7 and primalac=$userid and UNIX_TIMESTAMP(vrijeme)>$vrijeme");
	if (mysql_result($q30,0,0)>0) {
		?>
		<img src="images/newmail.gif" id="newmail" width="450" height="188" style="position:absolute;visibility:hidden" onload="newmail_show();">
		<script language="javascript">
		var mywidth,myheight;
		if (window.innerWidth && window.innerHeight) {
			mywidth=window.innerWidth;
			myheight=window.innerHeight;
		} else if (document.body.clientWidth && document.body.clientHeight) {
			mywidth=document.body.clientWidth;
			myheight=document.body.clientHeight;
		}

		var flashes=0;
		function newmail_show() {
			var newmail = document.getElementById('newmail');
			newmail.style.visibility='visible';
			newmail.style.top=-188;
			newmail.style.left=mywidth/2-225;
			setTimeout("newmail_scroll()",10);
		}
		function newmail_scroll() {
			var newmail = document.getElementById('newmail');
			if (parseInt(newmail.style.top) < myheight/2-94) {
				newmail.style.top=parseInt(newmail.style.top)+1;
				setTimeout("newmail_scroll()",10);
			} else {
				setTimeout("newmail_flash()",500);
			}
		}
		function newmail_flash() {
			var newmail = document.getElementById('newmail');
			if (newmail.style.visibility=='visible')
				newmail.style.visibility='hidden';
			else
				newmail.style.visibility='visible';
			flashes++;
			if (flashes<=10) setTimeout("newmail_flash()",500);
		}
		</script>
		<?
	}
}


// Standardne greske
if ($greska != "") {
	niceerror($greska);
	if ($sta=="") 
		zamgerlog("index.php greska: $greska $login ".my_escape($_REQUEST['sta']),3);
	else
		zamgerlog("index.php greska: $greska $login $sta",3);
}

// Poruka greške za modul
if ($found != 1 && $sta != "") {
	niceerror("Modul $sta još uvijek nije napravljen.");
	zamgerlog("pristup nepostojecom modulu $sta",3);
}

// Default moduli za uloge, u slučaju da modul nije pronađen
if ($found != 1) {
	if ($user_siteadmin) {
		$sta = "admin/intro";
	} else if ($user_studentska) {
		$sta = "studentska/intro";
	} else if ($user_nastavnik) {
		$sta = "saradnik/intro";
	} else if ($user_student) {
		$sta = "student/intro";
	} else {
		$sta = "public/intro";
	}
	$staf = str_replace("/","_",$sta);
}


// Promjena uloge korisnika
if ($userid>0) {
	if ($user_student && !strstr($sta,"student/"))
		print "<a href=\"?sta=student/intro\">Studentska stranica</a><br/>\n";
	if ($user_nastavnik && !strstr($sta,"saradnik/") && !strstr($sta,"nastavnik/"))
		print "<a href=\"?sta=saradnik/intro\">Spisak predmeta i grupa</a><br/>\n";
	if ($user_studentska && !strstr($sta,"studentska/"))
		print "<a href=\"?sta=studentska/intro\">Studentska služba</a><br/>\n";
	if ($user_siteadmin && !strstr($sta,"admin/"))
		print "<a href=\"?sta=admin/intro\">Site admin</a><br/>\n";
}


// Polje sa imenom i linkovima na inbox, profil i odjavu
if ($userid>0) {
	user_box();
	zamgerlog(urldecode(genuri()),1); // nivo 1 = posjet stranici
}

// Prikaz modula uglavljenog u template
include ("$sta.php");


// Prikaz menija specificnih za odredjene grupe modula
if (strstr($sta,"nastavnik/") || strstr($sta,"admin/"))
	malimeni("$staf();");
else if (strstr($sta,"studentska/"))
	horizontalni_meni("$staf();");
else if (strstr($sta,"student/"))
	studentski_meni("$staf();");
else
	eval("$staf();");


?>

	</td></tr>
</table>

<p>&nbsp;</p>
<p>&nbsp;</p>
<p align="center">Copyright (c) 2006-2010 Vedran Ljubović i drugi<br/>Elektrotehnički fakultet Sarajevo</p>

</body>
</html>
<?
	dbdisconnect();
?>
